TEL-6616: Fix generation of crypto keys.

dev-ryanc opened the pull request:
- Few cases where crypto variable may contain the whole crypto line or just key.

dev-ryanc committed:
TEL-6616: Fix generation of crypto keys.
 - Few cases where crypto variable may contain the whole crypto line or just key.

damirn commented on src/switch_core_media.c:
```diff
@@ -2103,11 +2118,11 @@ SWITCH_DECLARE(int) switch_core_session_check_incoming_crypto(switch_core_sessio
 			switch_channel_set_variable_printf(session->channel, "srtp_remote_audio_crypto_type", "%s", switch_core_media_crypto_type2str(ctype));
 		} else if (engine->type == SWITCH_MEDIA_TYPE_VIDEO) {
 			switch_channel_set_variable(session->channel, "srtp_remote_video_crypto_key", crypto);
-			switch_channel_set_variable_printf(session->channel, "srtp_remote_audio_crypto_tag", "%d", crypto_tag);
```
👍

damirn commented on src/switch_core_media.c:
```diff
@@ -1688,20 +1690,33 @@ static void switch_core_session_get_recovery_crypto_key(switch_core_session_t *s
 		if (ct) {
 			const char *local_crypto_key = NULL;
+			const char *inline_ptr = NULL;
+			const char *remote_key_material = NULL;
```
interesting name 'material' :)

damirn reviewed (approved):
nice!

dev-ryanc committed:
TEL-6616: Handle extracting key material with MKI
 - Handle scenario when key sizes change
 - Restore SDP session id and session version during recovery instead of generate new one.

damirn commented on src/switch_core_media.c:
```diff
@@ -1688,32 +1766,44 @@ static void switch_core_session_get_recovery_crypto_key(switch_core_session_t *s
-			local_crypto_key = switch_channel_get_variable(session->channel, lastkeyvar);
 			if (local_crypto_key) {
-				engine->ssec[engine->crypto_type].local_crypto_key = switch_core_session_strdup(session, local_crypto_key);
+				local_crypto_key = extract_crypto_key_material(local_crypto_key, local_key_buf, sizeof(local_key_buf));
```
it seems that `local_key_buf` is not being used?

dev-ryanc replied:
local_key_buf is just for buffer storage. The buffer will be the return value in the extract_crypto_key_material.
